### PR TITLE
168: Check that we receive unique time points per subject

### LINKED ISSUE #168: (Abrupt) sample size problem
**Summary**

Just change n from 700 to 800
```r

library(tidyr)
library(mmrm)

artificial=data.frame(USUBJID=sample(1:400,2528,replace=T),Visitn=sample(1:6,2528,replace=T),CD=sample(c(1,0),2528,replace=T),Y=rnorm(2528,mean=40,sd=10))

hist(artificial$Y)


msl=artificial
n=700


msl$Visitn=as.factor(msl$Visitn)

msl$USUBJID=as.factor(msl$USUBJID)
msl$CD=as.factor(msl$CD)

msl_test=msl[sample(1:nrow(msl),n,replace=F),]

fit <- mmrm(
  formula = Y ~ CD  + ad(Visitn | USUBJID),
  data = msl_test
)



```

**R session info**

```sh

# R -e "utils::sessionInfo()" output goes here

```

R version 4.2.1 (2022-06-23)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.6

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mmrm_0.1.5  tidyr_1.2.0

loaded via a namespace (and not attached):
 [1] pillar_1.8.1        compiler_4.2.1      tools_4.2.1         digest_0.6.29       evaluate_0.16       lifecycle_1.0.3     tibble_3.1.8       
 [8] checkmate_2.1.0     nlme_3.1-157        lattice_0.20-45     pkgconfig_2.0.3     rlang_1.0.6         DBI_1.1.3           cli_3.4.1          
[15] rstudioapi_0.13     yaml_2.3.5          parallel_4.2.1      mvtnorm_1.1-3       xfun_0.32           fastmap_1.1.0       dplyr_1.0.10       
[22] stringr_1.4.1       knitr_1.39          generics_0.1.3      vctrs_0.5.0         grid_4.2.1          tidyselect_1.2.0    glue_1.6.2         
[29] R6_2.5.1            fansi_1.0.3         Rdpack_2.4          rmarkdown_2.15      purrr_0.3.4         magrittr_2.0.3      emmeans_1.8.1-1    
[36] backports_1.4.1     htmltools_0.5.3     rbibutils_2.2.9     assertthat_0.2.1    xtable_1.8-4        numDeriv_2016.8-1.1 utf8_1.2.2         
[43] estimability_1.4.1  stringi_1.7.8

Gian & Carlos

### PR DESCRIPTION
**Pull request by danielinteractive**
closes #168

### CODE REVIEW

File: vignettes/algorithm.Rmd [hunk 1]
```diff
@@ -157,7 +157,7 @@ For example, for an unstructured covariance matrix, $\theta$ has $k = G * m(m+1)
 
 #### Spatial covariance matrix
 
-Under spatial covariance structure, the covariance matrix is different for all subjects, if they have different visit timepoints.
+Under spatial covariance structure, the covariance matrix is different for all subjects, if they have different time points.
```
Comment (kkmann) [inline vignettes/algorithm.Rmd:160]: suggestion:
"A spatial covariance structure can model individual-specific visit times. An individuals covariance matrix is then a function of both the population-level covariance parameters (specific to the chosen structure) and the individuals visit times."

ps: I know sas calls those spatial covariance structures but this just shows how confusing that terminology is x)

File: R/fit.R [hunk 2]
```diff
@@ -236,10 +236,17 @@ mmrm_control <- function(optimizer = stats::nlminb,
...
 #' `FEV1 ~ RACE + SEX + ARMCD * AVISIT + us(AVISIT | USUBJID)`
 #' so specifies response and covariates as usual, and exactly one special term
-#' defines which covariance structure is used and what are the visit and
-#' subject variables.
+#' defines which covariance structure is used and what are the time point and
+#' subject variables. The covariance structures in the formula can be
+#' found in [`covariance_types`].
 #'
-#' The covariance structures in the formula can be found in [`covariance_types`].
+#' Please note that the time points have to be unique for each subject. That is,
```
Comment (kkmann) [inline R/fit.R:243]: see other comments, quite verbous.

File: vignettes/introduction.Rmd [hunk 3]
```diff
@@ -105,6 +105,19 @@ fit_cs <- mmrm(
 print(fit_cs)
 ```
 
+Please note that the time points have to be unique for each subject. That is,
+there cannot be time points with multiple observations for any subject.
+The rationale is that these observations would need to be correlated, but it
+is not clear from the model definition what this correlation would be.
+If you e.g. have multiple assessments for one visit, then please create a
```
[the remark this answers is not in the document]
Comment (danielinteractive) [reply vignettes/introduction.Rmd:112]: Right, we could do that correctly with a more complex covariance structure framework here (and still keep independence of subjects which we cannot give up easily). Thanks Kevin! Simplified / shortened now.

File: vignettes/algorithm.Rmd (hunk 1 above)
Comment (danielinteractive) [reply vignettes/algorithm.Rmd:160]: done :-)

**Pull request merged**